src/frontends/tensorflow/src/op/dynamic_partition.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <limits>
+
+#include "op_table.hpp"
+#include "openvino/opsets/opset9.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::opset9;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_dynamic_partition_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"DynamicPartition"});
+    auto data = node.get_input(0);
+    auto partitions = node.get_input(1);
+    auto partitions_type = partitions.get_element_type();
+
+    // normalize partitions input since it can be a scalar or 1D tensor
+    auto new_parts_shape = make_shared<Constant>(element::i64, Shape{1}, -1);
+    auto norm_partitions = make_shared<Reshape>(partitions, new_parts_shape, true);
+
+    // retrieve num_partitions attribute
+    auto num_partitions = node.get_attribute<int64_t>("num_partitions");
+
+    // compute how many slices are collected for each partition
+    // that will be used as split_legths
+    auto start = make_shared<Constant>(partitions_type, Shape{}, 0);
+    auto end = make_shared<Constant>(partitions_type, Shape{}, num_partitions);
+    auto step = make_shared<Constant>(partitions_type, Shape{}, 1);
+    auto range_num_parts = make_shared<Range>(start, end, step, partitions_type);
+
+    auto unsqueeze_axis1 = make_shared<Constant>(element::i64, Shape{1}, 0);
+    auto unsqueeze_partitions = make_shared<Unsqueeze>(norm_partitions, unsqueeze_axis1);
+    auto unsqueeze_axis2 = make_shared<Constant>(element::i64, Shape{1}, 1);
+    auto unsqueeze_range = make_shared<Unsqueeze>(range_num_parts, unsqueeze_axis2);
+
+    auto mask = make_shared<Equal>(unsqueeze_range, unsqueeze_partitions);
+    auto mask_0_1 = make_shared<Select>(mask,
+                                        make_shared<Constant>(partitions_type, Shape{1}, 1),
+                                        make_shared<Constant>(partitions_type, Shape{1}, 0));
+    auto reduction_axis = make_shared<Constant>(element::i64, Shape{1}, 1);
+    auto split_legths = make_shared<ReduceSum>(mask_0_1, reduction_axis);
+
+    // for stable sorting using TopK operation, we have to re-scale partition indices by the formula:
+    // partition = partition * scale + partition_ind, where delta = max_int / num_partitions
```

Comment:
`..., where delta...` I don't see delta in the formula.

Reply:
It should be `scale`:). I will fix it in the next PR for `Unique` if no other concerns